File: program/program/implement-jump-search/ImplementJumpSearch.kt
import kotlin.math.sqrt

fun jumpSearch(arr: IntArray, x: Int): Int? {
    val n = arr.size
    var blockSize = sqrt(n.toDouble()).toInt()
    var prev = 0
    
    while (arr[minOf(blockSize, n) - 1] < x) {
        prev = blockSize
        blockSize += sqrt(n.toDouble()).toInt()
        if (prev >= n) {
            return null
        }
    }
    
    while (arr[prev] < x) {
        prev++
        if (prev == minOf(blockSize, n)) {
            return null
        }
    }
    
    if (arr[prev] == x) {
        return prev
    }
    
    return null
}

fun main() {
    // Take input from the user
    println("Enter the sorted array elements separated by spaces:")
    val inputArray = readLine()?.split(" ")?.map { it.toInt() }
    if (inputArray != null) {
        val list = inputArray.toIntArray()

        println("Enter the value to search:")
        val value = readLine()?.toInt()

        if (value != null) {
            val index = jumpSearch(list, value)
            if (index != null) {
                println("Output: $index")
            } else {
                println("Output: not found")
            }
        } else {
            println("Invalid input for the value.")
        }
    } else {
        println("Invalid input for the array.")
    }
}

File: program/program/implement-jump-search/ImplementJumpSearch.swift

import Foundation

func jumpSearch(_ arr: [Int], _ x: Int) -> Int? {
    let n = arr.count
    var blockSize = Int(sqrt(Double(n))) // Changed to var
    var prev = 0
    
    while arr[min(blockSize, n) - 1] < x {
        prev = blockSize
        blockSize += Int(sqrt(Double(n)))
        if prev >= n {
            return nil
        }
    }
    
    while arr[prev] < x {
        prev += 1
        if prev == min(blockSize, n) {
            return nil
        }
    }
    
    if arr[prev] == x {
        return prev
    }
    
    return nil
}

// Take input from the user
print("Enter the sorted array elements separated by spaces:")
if let input = readLine() {
    let list = input.split(separator: " ").compactMap { Int($0) } // Corrected
    print("Enter the value to search:")
    if let valueInput = readLine(), let value = Int(valueInput) {
        if let index = jumpSearch(list, value) {
            print("Output:", index)
        } else {
            print("Output: not found")
        }
    } else {
        print("Invalid input for the value.")
    }
} else {
    print("Invalid input for the array.")
}
